### ISSUE TITLE
[BUG] IPTABLES - need to check prior to adding rule if it exists already - iptables is not idempotent

### ISSUE THREAD

**Issue by MorningLightMountain713** [RunOnFlux/flux#1213]
This needs to be fixed asap.

Every time the function is being run, it's adding another rule. Need to check the rules first and only add it if it doesn't exist

```
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -j DROP
-A FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
-A FORWARD -d 172.16.32.0/24 -i docker0 -m state --state RELATED,ESTABLISHED -j ACCEPT
```

**Comment by MorningLightMountain713** [comment]
you can check if a rule exists with the `-C` flag

Example:

```
davew@chud:~$ sudo iptables -C FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j ACCEPT
ACCEPT  udp opt -- in docker0 out *  0.0.0.0/0  -> 172.16.32.0/24   udp dpt:53
davew@chud:~$ echo $?
0
davew@chud:~$ sudo iptables -C FORWARD -d 172.16.32.0/24 -i docker0 -p udp -m udp --dport 53 -j DROP
iptables: Bad rule (does a matching rule exist in that chain?).
davew@chud:~$ echo $?
1
davew@chud:~$
```

**Comment by MorningLightMountain713** [comment]
I can patch this if you want a pull...

**Comment by TheTrunk** [comment]
Hi, Yes please if you can create a PR that also includes a function that cleans up those rules that would be perfect 💯

**Comment by MorningLightMountain713** [comment]
fixed by #1234
